### PR TITLE
Adding ckan client api,

### PR DESCRIPTION
**Pull request by willdoran**
This pull request makes the following changes:
- Adding ckan laravel client
- The client is not lumen compatible so I had to add a work around to deal with the use of the config_path() function by autoloading a helper file.

Test checklist:
- [ ]

- [ ] I certify that I ran my checklist

Fixes ushahidi/platform#2881

Ping @ushahidi/platform

### CODE REVIEW

**Comment by willdoran** [comment]
@rjmackay @rowasc Do you have a better idea for working around the use of config_path(), we could either submit a PR to make their library lumen compatible or we could fork it or we could go with this, this is likely to crop up again though and this might be the best work around though we'd have to track which libs had the problem.

File: config/ckan_api.php [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CKAN_API_URL=""
+CKAN_API_KEY=""
```
Comment (rowasc) [inline config/ckan_api.php]: this needs to come from the database not from the .ENV since it's set up by each user

File: composer.json [hunk 2]
```diff
@@ -47,7 +47,8 @@
         "nexmo/client": "^1.1",
         "sentry/sentry-laravel": "^0.8.0",
         "laravel/passport": "^4.0",
-        "giggsey/libphonenumber-for-php": "^8.9"
+        "giggsey/libphonenumber-for-php": "^8.9",
```
Comment (rowasc) [inline composer.json]: why do we need this again?

Comment (willdoran) [reply composer.json]: no idea, I can remove it.

File: config/ckan_api.php (hunk 1 above)
Comment (willdoran) [reply config/ckan_api.php]: That is going to be more complex then because I don't think this client supports that but I'll look.

Comment (rowasc) [reply config/ckan_api.php]: cool. For context: this key is the one the user provides and you save in the user settings hdx.api_key config value

**Comment by willdoran** [comment]
It's possible this client isn't going to work because it instantiates the API details(url, key) only from config :(

**Comment by willdoran** [comment]
@rowasc I think this is correct, could you take a look and double check this does what you want. I included some example code in the HXL controller I can remove it before merging but it's just so you can see if it will do all you need. 

It's relatively simple, we import the client, pass it a guzzle client and then use it directly. So long as the config is formatted as in the example it should work as expected.

It will also return sensible errors, though those come directly from HDX we just pass them through.

**Comment by rowasc** [comment]
@willdoran I think this is OK (after commenting out the create call) . We will need some validation on our side because the CKAN API implementation in HDX fails at validating some things and just results in 500 errors when it should say 422:something , but this is what we need to start coding the integration so I think it's ok.

**Pull request merged**